src/processor/LakeOfFHIR.ts

```diff
@@ -95,4 +97,22 @@ export class LakeOfFHIR implements utils.Fishable {
     this.docs.forEach(d => defs.add(d.content));
     return defs.fishForMetadata(item, ...types);
   }
+
+  /**
+   * Removes any definitions from this.docs that have the same resourceType and id as a previous definition.
+   * Logs an error when it finds a duplicate
+   */
+  removeDuplicateDefinitions() {
+    this.docs = uniqWith(this.docs, (a, b) => {
+      const isDuplicate =
+        a.content.id === b.content.id && a.content.resourceType === b.content.resourceType;
+      if (isDuplicate) {
+        logger.error(
+          `Encountered a definition with the same resourceType and id as a previous definition: file "${a.path}" has resourceType ${a.content.resourceType} and id ${a.content.id}. ` +
+            'FHIR definitions should have unique resourceType and id. The duplicate will not be processed by GoFSH.'
+        );
+      }
+      return isDuplicate;
+    });
```

Comment:
Two questions/suggestions:

1. You list the path for `a` -- which almost makes it sound like `a` is the duplicate that will be tossed.  But I think `b` is the duplicate that is tossed, right?  If so, maybe it makes sense to tell the user the path of the thing we're tossing?  Or both paths?

2. If this happens a lot, that could be a lot of errors.  Since this is done in a loop (kind of), might it make sense just to collect one massive error something like:
> Encountered 17 definitions with the same resourceType and id as a previous definition.  FHIR definitions should be unique by resourceType and id.  The following duplicate definitions will not be processed by GoFSH:
> - /path/to/foo.xml (StructureDefinition/foo) matches /path/to/foo.json
> - /path/to/bar.xml (StructureDefinition/bar) matches /path/to/bar.json
> - ...

What do you think about that suggestion, @jafeltra?

Reply:
Yeah I was actually wondering about the number of errors, but when I was wondering that, I must have forgotten about the XML processing that would make this happen more often. I don't mind collecting a running tally and logging once.

As for your first point, I actually thought `a` was the duplicate that is being tossed, but I can double check. But maybe if I log the paths like you did in your example it doesn't matter quite as much.

Reply:
I think you'll still need to know if the duplicate is `a` or `b` in order to create the list / error message correctly.  I don't think the lodash doc is super clear about it, but I think maybe I misunderstood it and you may be right about `a` being the duplicate.

Reply:
I tried it with the REPL in lodash docs, and when I did this:
```
var objects = [{ 'x': 1, 'y': 'first' }, { 'x': 2, 'y': 1 }, { 'x': 1, 'y': 'second' }];
_.uniqWith(objects, (a, b) => a.x === b.x);
```
I got this:
```
[{x: 1, y: "first"}, {x: 2, y: 1}]
```
So I think that shows the second value is tossed right?

Reply:
Oh wait maybe that doesn't answer our question I see.

Reply:
Right...  at the point in the comparator where we're checking for the dup, we only have `a` and `b`.  I guess we could loop through the original array to find each and see which has the lower index, but... that seems like it should be unnecessary (I hope).

Reply:
I don't know that it truly matters much, but it might help a little when understanding the error message. That said, I think it is working as we'd expect, and this is what the message looks like now with f77d157:

![image](https://user-images.githubusercontent.com/30803904/107263695-10def400-6a10-11eb-83ba-534e08a65320.png)

Reply:
Oh no, GitHub did the thing were it didn't show me all the new conversation. Let me read and ignore my message if there are obvious contradictions.